docs/content/mongocxx-v3/client-side-encryption.md

```diff
@@ -0,0 +1,175 @@
++++
+title = "Client Side Encryption with mongocxx"
+[menu.main]
+  identifier = 'mongocxx-cse'
+  weight = 11
+  parent='mongocxx3'
++++
+
+## Client Side Encryption
+
+New in MongoDB 4.2 client side encryption allows administrators and developers to encrypt specific data fields in addition to other MongoDB encryption features.
+
+With field level encryption, developers can encrypt fields client side without any server-side configuration or directives. Client-side field level encryption supports workloads where applications must guarantee that unauthorized parties, including server administrators, cannot read the encrypted data.
+
+For an overview of client-side encryption, please read [the official MongoDB documentation in the manual](https://docs.mongodb.com/manual/core/security-client-side-encryption/).
+
+## Installation
+
+### libmongocrypt
+
+Client-side encryption relies on a C library called `libmongocrypt` to do the heavy lifting encryption work. This dependency is managed by the C driver.  As long as the C driver installation is up-to-date, this dependency should be managed internally.
+
+### mongocryptd
+
+Automatic client-side encryption relies upon a new binary called `mongocryptd` running as a daemon while the driver operates.  This binary is only available with MongoDB Enterprise.
+
+`mongocryptd` can either be started separately from the driver, or left to spawn automatically when encryption is used.
+
+To run mongocryptd separately, pass the `mongocryptdBypassSpawn` flag to the client's auto encryption options:
+
+```c++
+auto mongocryptd_options = make_document(kvp("mongocryptdBypassSpawn", true));
+
+options::auto_encryption auto_encrypt_opts{};
+auto_encrypt_opts.extra_options({mongocryptd_options.extract()});
+```
+
+If the mongocryptd binary is on the current path, the driver will be able to spawn it without any custom flags.  However, if the mongocryptd binary is on a different path, set the path with the `mongocryptdSpawnPath` option:
+
+```c++
+auto mongocryptd_options = make_document(kvp("mongocryptdSpawnPath", "path/to/mongocryptd"));
+
+options::auto_encryption auto_encrypt_opts{};
+auto_encrypt_opts.extra_options({mongocryptd_options.extract()});
+```
+
+## Examples
+
+### Automatic Encryption with a Local Key
+
+The following is sample code that assumes the key and schema have already been created in MongoDB. The example uses a local key, however using AWS Key Management Service is also an option. The data in the `encryptedField` field is automatically encrypted on the insert and decrypted when using find on the client side.
+
+These code snippets come from `examples/mongocxx/client_side_encryption.cpp`, which ships with the driver.
+
+Generate an encryption key:
```

Comment:
Note, I recently [commented](https://jira.mongodb.org/browse/DRIVERS-752?focusedCommentId=3297526&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3297526 ) on DRIVERS-752. Unfortunately the link to the Java docs was to an older version, which was later superseded by the example updates of DRIVERS-782. Those updates are mostly additive, so I think we can wait to do them until the associated C++ driver ticket, CXX-1887, but I wanted to give heads up in case you refer back to the Java driver docs and see they no longer match.

Reply:
Noted.